### PR TITLE
Create a clone of local segements size map used for Remote Segment Stats until sync to remote completes

### PR DESCRIPTION
**Pull request by linuxpi**
### Description
- Clones local segment files map stored in segment tracker during upload operations to ensure immutability of the map for that scope
- map in segment tracker could be updated while segments are uploaded to remote by another refresh.

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/11025 and https://github.com/opensearch-project/OpenSearch/issues/9774

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java [hunk 1]
```diff
@@ -306,7 +306,7 @@ public Map<String, Long> getLatestLocalFileNameLengthMap() {
      * @param segmentFiles     list of local refreshed segment files
      * @param fileSizeFunction function is used to determine the file size in bytes
      */
-    public void updateLatestLocalFileNameLengthMap(
+    public Map<String, Long> updateLatestLocalFileNameLengthMap(
```
Comment (ashking94) [inline server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java:309]: nit - update java doc

**Review by ashking94 — APPROVED** [review]
Please update on how many iterations we have run for asserting the fix in flakiness of this test.

**Comment by linuxpi** [comment]
> Please update on how many iterations we have run for asserting the fix in flakiness of this test.

1000 iterations. earlier issue used to popup within 300 iterations

File: server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java (hunk 1 above)
Comment (linuxpi) [reply server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java:314]: done

**Comment by linuxpi** [comment]
> ❌ Gradle check result for [7875161](https://github.com/opensearch-project/OpenSearch/commit/78751610cc9e78aa9fb88eac5c3d2af3a3f01def): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/32230/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?


[org.opensearch.cluster.MinimumClusterManagerNodesIT.testThreeNodesNoClusterManagerBlock](https://build.ci.opensearch.org/job/gradle-check/32230/testReport/junit/org.opensearch.cluster/MinimumClusterManagerNodesIT/testThreeNodesNoClusterManagerBlock/) https://github.com/opensearch-project/OpenSearch/issues/10006
    [org.opensearch.search.sort.FieldSortIT.testSimpleSorts {p0={"search.concurrent_segment_search.enabled":"true"}}](https://build.ci.opensearch.org/job/gradle-check/32230/testReport/junit/org.opensearch.search.sort/FieldSortIT/testSimpleSorts__p0___search_concurrent_segment_search_enabled___true___/)  https://github.com/opensearch-project/OpenSearch/issues/11875

flaky tests failing

**Comment by gbbafna** [comment]
Should we give https://github.com/opensearch-project/OpenSearch/issues/11908 a try out ? This is getting too complex .

**Comment by ashking94** [comment]
> Should we give #11908 a try out ? This is getting too complex .

Lets fix this flaky test and I (or whoever can) can attempt the simplification of the listener.

**Comment by gbbafna** [comment]
> > Should we give #11908 a try out ? This is getting too complex .
> 
> Lets fix this flaky test and I (or whoever can) can attempt the simplification of the listener.

Can we attempt it now (just POCs and finalizing approach) and then come back to this PR depending upon the complexity ? If we can achieve it now, we wouldn't be needing this PR at all .  If it is burning problem, we can always mute the test  and it can get resolved once we simplify this issue.

**Review by gbbafna — APPROVED** [review]

**Review by gbbafna — APPROVED** [review]

**Pull request merged**